### PR TITLE
feat: Add a method to log actions and clicks

### PR DESCRIPTION
**Pull request by prm-kevingrandon**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -56,6 +57,8 @@ type PropValues<T> = T extends "handleError"
   ? string
   : T extends "logImpression"
   ? (impression: Impression) => void
+  : T extends "logAction"
+  ? (impression: Impression) => void
```
Comment (prm-dan) [inline src/index.ts:61]: Ah, I should have brought up earlier.  They might have multiple action buttons on a single listing.  E.g. click, like, share.  We can attach impressionId to all of those actions.

**Review by prm-dan — COMMENTED** [review]
Sorry for the delay.  GitHub blocked my previous comment but I didn't see it.

File: src/index.ts (hunk 1 above)
Comment (prm-kevingrandon) [reply src/index.ts:61]: Sounds good, I've added support for passing in a custom impression ID on clicks.

File: src/index.ts [hunk 2]
```diff
@@ -165,8 +166,30 @@ export default Vue.extend({
...
+
+    logActionFunctor({ impressionId }: { impressionId?: string } = {}) {
+      if (!this.active) {
+        return;
+      }
+
+      const impression = this.makeImpression();
+      this.typedProp("logAction") &&
+        this.typedProp("logAction")({
+          ...impression,
```
Comment (prm-dan) [inline src/index.ts:188]: Action type != Impression type.
- `sourceType` is not valid.
- `actionType` is needed.
- For navigates, we can take some other options fields.

https://github.com/promotedai/promoted-snowplow-logger/blob/main/src/types/event.d.ts#L94

Comment (prm-kevingrandon) [reply src/index.ts]: Thanks, I've updated types for actions.

File: src/index.ts [hunk 3]
```diff
@@ -5,6 +5,94 @@ import Vue from "vue";
...
+export interface Action {
+  userInfo?: {
+    anonUserId?: string;
+    userId?: string;
+    isInternalUser?: boolean;
+  };
+  insertionId?: string;
+  contentId?: string;
+  impressionId: string;
+  actionType?: ActionTypeMap[keyof ActionTypeMap] | ActionTypeString;
```
Comment (prm-dan) [inline src/index.ts:86]: The backend also supports

```
customActionType: string,
```

It'd be good to add that too.

File: src/index.ts [hunk 4]
```diff
@@ -165,8 +260,51 @@ export default Vue.extend({
...
       }
+
       this.typedProp("logImpression") && this.typedProp("logImpression")(impression);
     },
+
+    logActionFunctor({
+      impressionId,
+      elementId,
+      targetUrl,
+      actionType = 2,
```
Comment (prm-dan) [inline src/index.ts:271]: And add `customActionType` here too.

**Review by prm-dan — APPROVED** [review]

**Pull request merged**